packages/frontend/app/(dashboard)/course/[cid]/insights/context/InsightsContext.tsx

```diff
@@ -0,0 +1,80 @@
+'use client'
+
+import React, { createContext, useContext, useEffect, useState } from 'react'
+import { API } from '@/app/api'
+import { UserPartial, Role, QueuePartial } from '@koh/common'
+import { useCourse } from '@/app/hooks/useCourse'
+
+type InsightContextType = {
+  studentDetails?: {
+    students: UserPartial[]
+    totalStudents: number
+    page: number
+    setPage: (n: number) => void
+    search: string
+    setSearch: (s: string) => void
+  }
+  queueDetails?: QueuePartial[]
+}
+
+const InsightContext = createContext<InsightContextType>({})
+
+const useInsightContext = () => {
+  const context = useContext(InsightContext)
+  if (context == undefined) {
+    throw new Error(
+      'Cannot use InsightContext outside of an InsightContextProvider.',
+    )
+  }
+  return context
+}
+
+const InsightContextProvider: React.FC<{
+  courseId: number
+  children: React.ReactNode
+}> = ({ courseId, children }) => {
+  const { course } = useCourse(courseId)
+  const [students, setStudents] = useState<UserPartial[]>([])
+  const [totalStudents, setTotalStudents] = useState<number>(0)
+  const [page, setPage] = useState<number>(1)
+  const [search, setSearch] = useState<string>('')
+
+  useEffect(() => {
+    ;(async () => {
```

Comment:
what's mr `;` doing here?

Reply:
😬

Reply:
addressed! guess the linter didn't catch _this_ one